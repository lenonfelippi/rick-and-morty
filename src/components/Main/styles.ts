'use client'

import styled from 'styled-components'

export const Wrapper = styled.main`
  background-color: green;
  color: #fff;
  width: 100%;
  height: 100%;
`
