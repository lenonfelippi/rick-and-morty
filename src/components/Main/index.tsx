import * as S from './styles'

const Main = () => {
  return (
    <S.Wrapper>
      <h1>Testando o jest</h1>
    </S.Wrapper>
  )
}

export default Main
