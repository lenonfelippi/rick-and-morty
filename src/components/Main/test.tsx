import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('should render the heading', () => {
    // renderiza o componente
    const { container } = render(<Main />)
    // busca o elemento e verifica a existencia
    expect(
      screen.getByRole('heading', { name: /testando o jest/i })
    ).toBeInTheDocument()

    //gerar snapshot
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    // renderiza o componente
    const { container } = render(<Main />)

    // verifica se o bg esta correto
    expect(container.firstChild).toHaveStyle({ 'background-color': 'green' })
  })
})
