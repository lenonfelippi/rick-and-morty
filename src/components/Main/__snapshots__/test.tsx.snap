// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
<main
  class="sc-bdnyFh kFtGJI"
>
  <h1>
    Testando o jest
  </h1>
</main>
`;
